### PR TITLE
Update docs and add state for junos_static_routes

### PR DESCRIPTION
**Pull request by danielmellado**
This commit adds gathered state for junos_static_routes and updates the
documentation to the new resource modules versioning for collections.

### CODE REVIEW

**Comment by pabelanger** [comment]
recheck

File: plugins/modules/junos_static_routes.py [hunk 1]
```diff
@@ -29,14 +29,10 @@
...
-ANSIBLE_METADATA = {
-    "metadata_version": "1.1",
-    "status": ["preview"],
-    "supported_by": "network",
-}
+ANSIBLE_METADATA = {"metadata_version": "1.1", "supported_by": "network"}
 
 DOCUMENTATION = """module: junos_static_routes
-short_description: Manage static routes on Juniper JUNOS devices
+short_description: Static routes JUNOS Resource Module
```
Comment (samccann) [inline plugins/modules/junos_static_routes.py:35]:
```suggestion
short_description: Static routes Junos resource module
```

**Comment by danielmellado** [comment]
recheck

**Comment by danielmellado** [comment]
recheck

**Comment by danielmellado** [comment]
recheck

**Comment by pabelanger** [comment]
recheck

**Pull request merged**